packages/react-components/src/components/utils/SendBoxUtils.ts

```diff
@@ -37,14 +37,36 @@ export const isAttachmentUploadCompleted = (
   return !!attachmentsWithProgress?.find((attachment) => !attachment.error);
 };
 
+/* @conditional-compile-remove(rich-text-editor-image-upload) */
+/**
+ * Obtain any image nodes from the content DOM passed in
+ * returns a list of image IDs
+ * @internal
+ */
+export const inlineImageIds = (content?: string): { id: string; url: string }[] => {
+  if (!content) {
+    return [];
+  }
+  const document = new DOMParser().parseFromString(content, 'text/html');
+  const imageTags = document.querySelectorAll('img');
+  const ids: { id: string; url: string }[] = [];
+  imageTags.forEach((node) => {
+    const id = node.id;
+    const url = node.src;
+    if (id && url) {
+      ids.push({ id, url });
```

Comment:
Is there a reason to check id and url? I'm not quite sure whether we always have both. I kind of remember in some edge case we remove src

Reply:
That's good to know; updated.